src/main/java/org/opensearch/neuralsearch/processor/normalization/MinMaxScoreNormalizationTechnique.java

```diff
@@ -167,22 +225,182 @@ private float[] getMinScores(final List<CompoundTopDocs> queryTopDocs, final int
         return minScores;
     }
 
-    private float normalizeSingleScore(final float score, final float minScore, final float maxScore) {
+    private float normalizeSingleScore(final float score, final float minScore, final float maxScore, final LowerBound lowerBound) {
         // edge case when there is only one score and min and max scores are same
         if (Floats.compare(maxScore, minScore) == 0 && Floats.compare(maxScore, score) == 0) {
             return SINGLE_RESULT_SCORE;
         }
-        float normalizedScore = (score - minScore) / (maxScore - minScore);
-        return normalizedScore == 0.0f ? MIN_SCORE : normalizedScore;
+        if (!lowerBound.isEnabled()) {
+            return LowerBound.Mode.IGNORE.normalize(score, minScore, maxScore, lowerBound.getMinScore());
+        }
+        return lowerBound.getMode().normalize(score, minScore, maxScore, lowerBound.getMinScore());
+    }
+
+    private Optional<List<Pair<LowerBound.Mode, Float>>> getLowerBounds(final Map<String, Object> params) {
+        if (Objects.isNull(params) || !params.containsKey(PARAM_NAME_LOWER_BOUNDS)) {
+            return Optional.empty();
+        }
+
+        List<Pair<LowerBound.Mode, Float>> lowerBounds = new ArrayList<>();
+
+        Object lowerBoundsObj = params.get(PARAM_NAME_LOWER_BOUNDS);
+        if (!(lowerBoundsObj instanceof List<?> lowerBoundsParams)) {
+            throw new IllegalArgumentException("lower_bounds must be a List");
+        }
+
+        if (lowerBoundsParams.size() > MAX_NUMBER_OF_SUB_QUERIES) {
+            throw new IllegalArgumentException(
+                String.format(
+                    Locale.ROOT,
+                    "lower_bounds size %d should be less than or equal to %d",
+                    lowerBoundsParams.size(),
+                    MAX_NUMBER_OF_SUB_QUERIES
+                )
+            );
+        }
+
+        for (Object boundObj : lowerBoundsParams) {
+            if (!(boundObj instanceof Map)) {
+                throw new IllegalArgumentException("each lower bound must be a map");
+            }
+
+            @SuppressWarnings("unchecked")
+            Map<String, Object> lowerBound = (Map<String, Object>) boundObj;
+
+            try {
+                LowerBound.Mode mode = LowerBound.Mode.fromString(
+                    Objects.isNull(lowerBound.get(PARAM_NAME_LOWER_BOUND_MODE))
+                        ? ""
+                        : lowerBound.get(PARAM_NAME_LOWER_BOUND_MODE).toString()
```

Comment:
```suggestion
                LowerBound.Mode mode = LowerBound.Mode.fromString(
                    Objects.nonNull(lowerBound.get(PARAM_NAME_LOWER_BOUND_MODE))
                        ? lowerBound.get(PARAM_NAME_LOWER_BOUND_MODE).toString():""
```

Reply:
not sure what's the benefit of this change, and format is enforced by spotless, will keep my version